tsdb/head_append.go

```diff
@@ -352,6 +359,11 @@ func (a *headAppender) Append(ref storage.SeriesRef, lset labels.Labels, t int64
 		}
 	}
 
+	if v == 0 && s.lastTimestamp > t {
+		// Created timestamp that was already appended, skip.
+		return 0, nil
+	}
```

Comment:
@TheSpiritXIII @macxamin looking at this again, we can't assume it is a created timestamp just because value is zero.

I think we'll need another method, `AppendCreatedTimestamp()`, in the appender interface :/